design-patterns/src/mockGui.ts

```diff
@@ -0,0 +1,22 @@
+import { eventListeners, shipmentCallback, ShipmentState } from "./types";
+
+export interface GUI {
+  on: (eventType: string, callback: shipmentCallback) => void;
+  trigger: (eventType: string, state: ShipmentState) => void;
+}
+
+export class MockGUI implements GUI {
```

Comment:
Classic Observer design pattern or Publish-Subscribe pattern - Why is it MockGUI naming? just because types?